lib/visitor/strategy/default_strategy.dart

```diff
@@ -184,7 +205,101 @@ class DefaultStrategy implements IVisitor {
         visitor.anonymousId = null;
       }
     } else {
-      Flagship.logger(Level.ALL, "unAuthenticateVisitor method will be ignored in Bucketing configuration");
+      Flagship.logger(Level.ALL,
+          "unAuthenticateVisitor method will be ignored in Bucketing configuration");
     }
   }
+
+  @override
+  void cacheVisitor(String visitorId, String jsonString) {
+    visitor.config.visitorCacheImp?.cacheVisitor(visitor.visitorId, jsonString);
+  }
+
+  @override
+  // Called right at visitor creation, return a jsonString corresponding to visitor. Return a jsonString
+  void lookupVisitor(String visitoId) async {
+    visitor.config.visitorCacheImp
+        ?.lookupVisitor(visitor.visitorId)
+        .then((resultFromCache) {
+      if (resultFromCache.length != 0) {
+        // convert to Map
+        Map<String, dynamic> result = jsonDecode(resultFromCache);
+        // Retreive the json string stored in the visitor filed of this map.
+        if (result['visitor'] != null) {
+          VisitorCache cachedVisitor =
+              VisitorCache.fromJson(jsonDecode(result['visitor']));
+          Flagship.logger(Level.DEBUG,
+              'The cached visitor get through the lookup is ${cachedVisitor.toString()}');
+          // update the current visitor with his own cached data
+          // 1 - update modification Map<String, Modification> modifications
+          visitor.modifications
+              .addEntries(cachedVisitor.getModifications().entries);
+          if (visitor.config.decisionMode == Mode.BUCKETING) {
```

Comment:
Could we remove this if ? for instance by implementing a "null" function `updateAssignationHistory` event for decision API manager ?

Reply:
yes , no need to have an **if** here, i planned this function in the interface which is not implemented by the api_manager, yet